cmd/main_test.go

```diff
@@ -301,3 +307,122 @@ var _ = DescribeTable("TestRunAccessCheck", func(user string, namespace string,
+					},
+					Status: crt.WorkspaceStatus{
+						Namespaces: []crt.SpaceNamespace{
+							{
+								Name: "test-tenant",
+								Type: "default",
+							},
+						},
+					},
+				},
+				{
+					TypeMeta: metav1.TypeMeta{
+						Kind:       "Workspace",
+						APIVersion: gv,
+					},
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "test-tenant-2",
+					},
+					Status: crt.WorkspaceStatus{
+						Namespaces: []crt.SpaceNamespace{
+							{
+								Name: "test-tenant-2",
+								Type: "default",
+							},
+						},
+					},
+				},
+			}
+		})
+
+		It("Should return a WorkspaceList with test-tenant workspace and both namespaces in it", func() {
+			actualWorkspaces, err := getWorkspacesWithAccess(e, c, allNamespaces)
+			Expect(actualWorkspaces.Items).To(Equal(expectedWorkspaces))
+			Expect(err).NotTo(HaveOccurred(), "Unexpected error testing GetWorkspacesWithAccess")
+		})
+	})
+
+	Context("When workspace with only test-tenant namespace has all the necessary permissions", func() {
+		BeforeEach(func() {
+			gv = "v1alpha1"
+			namespaceNames := []string{"test-tenant", "test-tenant-3"}
+			for _, name := range namespaceNames {
+				ns, err := createNamespace(k8sClient, name)
+				Expect(err).NotTo(HaveOccurred(), fmt.Sprintf("Error while creating the namespace %s: %v", name, err))
+				allNamespaces = append(allNamespaces, ns)
+			}			
+			createRole(k8sClient, "test-tenant", "namespace-access", []string{"create", "list", "watch", "delete"})
+			createRoleBinding(k8sClient, "namespace-access-user-binding", "test-tenant", "user1@konflux.dev", "namespace-access")
+			createRoleBinding(k8sClient, "namespace-access-user-binding-3", "test-tenant-3", "user1@konflux.dev", "namespace-access")
+			expectedWorkspaces = []crt.Workspace{
+				{
+					TypeMeta: metav1.TypeMeta{
+						Kind:       "Workspace",
+						APIVersion: gv,
+					},
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "test-tenant",
+					},
+					Status: crt.WorkspaceStatus{
+						Namespaces: []crt.SpaceNamespace{
+							{
+								Name: "test-tenant",
+								Type: "default",
+							},
+						},
+					},
+				},
+			}
+		})
+
+		It("Should return a WorkspaceList with test-tenant workspace and only test-tenant namespace in it", func() {
+			actualWorkspaces, err := getWorkspacesWithAccess(e, c, allNamespaces)
+			Expect(actualWorkspaces.Items).To(Equal(expectedWorkspaces))
+			Expect(err).NotTo(HaveOccurred(), "Unexpected error testing GetWorkspacesWithAccess")
+		})
+	})	
+
+	Context("When no workspaces has all the necessary permissions", func() {
+		BeforeEach(func() {
+			gv = "v1alpha1"
+			namespaceNames := []string{"test-tenant", "test-tenant-2"}
+			for _, name := range namespaceNames {
+				ns, err := createNamespace(k8sClient, name)
+				Expect(err).NotTo(HaveOccurred(), fmt.Sprintf("Error while creating the namespace %s: %v", name, err))
+				allNamespaces = append(allNamespaces, ns)
+			}			
+			expectedWorkspaces = []crt.Workspace{}
```

Comment:
I'm not sure how that is the result we are getting, taking into considiration we did not clean up the resources we created on the other test cases